cpp/FileTransferAgent.cpp

```diff
@@ -248,6 +248,30 @@ void Snowflake::Client::FileTransferAgent::uploadFilesInParallel(std::string *co
     FileMetadata * metadata = &m_smallFilesMeta[i];
     m_executionResults->SetFileMetadata(&m_smallFilesMeta[i], resultIndex);
     metadata->overWrite = response.overwrite;
+
+    // workaround for incident 00212627
+    // Do not use thread pool when parallel = 1
+    if ((unsigned int)response.parallel <= 1)
```

Comment:
Can response.parallel ever be less than 1?

Reply:
@sfc-gh-kwagner The approved values are 1 to 99 with 1 no parallelism and 99 highest level and default 4.